packages/next/src/build/webpack/loaders/next-metadata-route-loader.ts

```diff
@@ -148,15 +151,18 @@ export async function generateStaticParams() {
 
   const code = `\
 import { NextResponse } from 'next/server'
-import * as _sitemapModule from ${JSON.stringify(resourcePath)}
+import * as userland from ${JSON.stringify(resourcePath)}
 import { resolveRouteData } from 'next/dist/build/webpack/loaders/metadata/resolve-route-data'
 
-const sitemapModule = { ..._sitemapModule }
+const sitemapModule = { ...userland }
 const handler = sitemapModule.default
 const generateSitemaps = sitemapModule.generateSitemaps
 const contentType = ${JSON.stringify(getContentType(resourcePath))}
 const fileType = ${JSON.stringify(getFilenameAndExtension(resourcePath).name)}
 
+${'' /* re-export the userland route configs */}
+export * from ${JSON.stringify(resourcePath)}
```

Comment:
Will this cause duplicate exports? Webpack warns if we re-export a GET function here.

Another question is, does this break static analyzation (if we have) for configs?

Reply:
> Will this cause duplicate exports? Webpack warns if we re-export a GET function here.

No since GET export is not supported in metadata routes, only "default" export and few "metadata" export for image metadata routes.

> Another question is, does this break static analyzation (if we have) for configs?

Current structure is going through like this: user code -> metadata route loader -> app route metadata wrapper (created by @wyattjoh), there we will check the final exported value.